frontend/src/layouts/AppLayout.jsx

```diff
@@ -1,25 +1,32 @@
-import { useContext, useEffect, useState } from "react";
-import { Navigate, Outlet } from "react-router-dom";
+import { useContext } from "react";
+import { Navigate, Outlet, useNavigate } from "react-router-dom";
 import { AuthContext } from "../contexts/AuthContext";
+import Cookies from "js-cookie";
+import { logout } from "../api/auth";
 
 export default function AppLayout() {
-  const token = "";
   const { checkAuth } = useContext(AuthContext);
-  const [isAuthUrl, setIsAuthUrl] = useState(null);
+  const navigate = useNavigate();
+  const cookies = Cookies.get("user");
 
-  useEffect(() => {
-    const navigateURL = checkAuth(token);
-    if (navigateURL) {
-      setIsAuthUrl(navigateURL);
-    }
-  }, [token, checkAuth]);
-  if (isAuthUrl) {
-    return <Navigate to={isAuthUrl} />;
+  if (!checkAuth(cookies)) {
+    return <Navigate to="/auth/login" />;
   }
-  console.log(isAuthUrl);
+
+  const handleLogout = async () => {
+    const res = await logout();
+    console.log(res.message);
+    navigate("/");
+  };
+
```

Comment:
Once you have made changes to to the AuthContext, use isAuthenticated state to conditionally navigate to /auth/login

Reply:
Can't fully understand what do you mean

Reply:
In the AppLayout, based on the isAuthenticated value imported from the context, navigate to login page.
Here is a rough snippet on how to do that. 
```js
function AppLayout({children}) {
 const { isAuthenticated } = useContext(AuthContext);
 /*
  Other layout logic if necessary
 */
 if (!isAuthenticated) return <Navigate to="/auth/login" />;

 // if user isAuthenticated, return the layout
 return <div><Outlet /></div>;
}
```